src/app/pages/layout/layout.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, inject } from '@angular/core';
+import { MatIconRegistry } from '@angular/material/icon';
+import { DomSanitizer } from '@angular/platform-browser';
+import { SvgEnum } from 'src/app/enum';
+import { LayoutService } from './layout.service';
+
+@Component({
+  selector: 'app-layout',
+  templateUrl: './layout.component.html',
+  styleUrls: ['./layout.component.scss'],
+})
+export class LayoutComponent {
+  sidenaveOpen = false;
+  iconRegistry = inject(MatIconRegistry);
+  domSanitizer = inject(DomSanitizer);
+  layoutService = inject(LayoutService);
```

Comment:
you can directly use this instead of sidenaveOpen

Reply:
done